tests/integration/test_authd/test_authd_force_options.py

```diff
@@ -0,0 +1,252 @@
+
+    time.sleep(1)
+    # Start Wazuh
+    control_service('start', daemon='wazuh-authd')
+
+    """Wait until agentd has begun"""
+
+    def callback_agentd_startup(line):
+        if 'Accepting connections on port 1515' in line:
+            return line
+        return None
+
+    log_monitor = FileMonitor(LOG_FILE_PATH)
+    log_monitor.start(timeout=30, callback=callback_agentd_startup)
+    time.sleep(1)
+
+
+def check_client_keys_file(response):
+    try:
+        with open(client_keys_path) as client_file:
+            client_lines = client_file.readlines()
+            for line in client_lines:
+                # discard endline because response finalize with "'\n\n" character and client.keys only with "\n"
+                if line[:-1] in response:
+                    client_file.close()
+                    return True
+    except IOError as exception:
+        raise
+    return False
+
+
+def send_message(message):
+    address, family, connection_protocol = receiver_sockets_params[0]
+    SSL_socket = SocketController(address, family=family, connection_protocol=connection_protocol)
+    try:
+        SSL_socket.open()
+    except ssl.SSLError as exception:
+        # We did not expect this error, fail test
+        raise
+    SSL_socket.send(message, size=False)
+    response = SSL_socket.receive().decode()
+    SSL_socket.close()
+    return response
+
+
+# Initial clean client_keys file
+# Stop Wazuh
+control_service('stop')
+
+# reset_client_keys
+clean_client_keys_file()
+
+# Start Wazuh
+control_service('start')
+
+"""Wait until agentd has begun"""
+
+
+def callback_agentd_startup(line):
+    if 'Accepting connections on port 1515' in line:
+        return line
+    return None
+
+
+log_monitor = FileMonitor(LOG_FILE_PATH)
+log_monitor.start(timeout=30, callback=callback_agentd_startup)
+
+
+# @pytest.mark.parametrize('test_case', [case['test_case'] for case in ssl_configuration_tests])
+def test_authd_force_options(get_configuration, configure_environment, configure_sockets_environment):
+    """Check that every input message in authd port generates the adequate output
+
+    Parameters
+    ----------
+    test_case : list
+        List of test_cases, dict with following keys:
+            - input: message that will be tried to send to the manager
+            - output: expected response
+            - insert_prev_agent: yes or no (for duplicated ip or name cases)
+                1) if insert_prev_agent_custom is present: previous input message is overwrite by the custom message
+                    (insert_prev_agent_custom: "OSSEC A:'user0' IP:'10.10.10.10'")
+                2) if insert_prev_agent_custom is not present: send the masage equals to input
+            - insert_random_pass_in_query:
+              "yes" if is needed add random pass to input query (for register with random pass cases)
+            - insert_hostname_in_query:
+              "yes" if is present add host name to input message
+    """
```

Comment:
The docstring should follow google style:  https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
The test description was updated